src/devices/Ssd1351/Ssd1351.cs

```diff
@@ -0,0 +1,312 @@
+                        _colorDepth = (ColorDepth)((data[0] >> 6) & 0x03);
+                        break;
+                }
+
+            }
+
+        }
+
+        /// <summary>
+        /// Send a command to the the display controller.
+        /// </summary>
+        /// <param name="data">Byte array to send as a combination of command in the first byte
+        /// and following with parameters for the command.</param>
+        public void SendCommand(byte[] data)
+        {
+            Span<byte> buffer = data;
+
+            if(data == null || data.Length == 0)
+            {
+                throw new ArgumentNullException(nameof(data));
+            }
+
+            if (data.Length == 0)
+            {
+                throw new ArgumentException("Data array is empty", nameof(data));
+            }
+
+            SendCommand((SSD1351Command)data[0], buffer.Slice(1).ToArray());
+        }
+
+        /// <summary>
+        /// Send data to the display controller.
+        /// </summary>
+        /// <param name="data">The data to send to the display controller.</param>
+        public void SendData(byte[] data)
+        {
+            const int StackThreshold = 512;
+
+            if (data == null)
+            {
+                throw new ArgumentNullException(nameof(data));
+            }
+
+            Span<byte> buffer = data.Length < StackThreshold ? stackalloc byte[data.Length + 1] : new byte[data.Length + 1];
+
+            buffer[0] = 0x40; // Control byte.
+            data.CopyTo(buffer.Slice(1));
+            SendSPI(buffer);
+        }
+
+        // Write out a COMMAND to the display
+        /// <summary>
+        /// Write a block of data to the SPI device
+        /// </summary>
+        /// <param name="data">The data to be sent to the SPI device</param>
+        /// <param name="blnIsCommand">A flag indicating that the data is really a command when true or data when false.</param>
+        private void SendSPI(Span<byte> data, bool blnIsCommand = false)
+        {
+            int index = 0;
+            int len;
+
+            // set the DC pin to indicate if the data being sent to the display is DATA or COMMAND bytes.
+            _gpioDevice.Write(_dcPinId, blnIsCommand ? PinValue.Low : PinValue.High);
+
+            // write the array of bytes to the display. (in chunks of SPI Buffer Size)
+            do
+            {
+                // calculate the amount of spi data to send in this chunk
+                len = Math.Min(data.Length - index, _spiBufferSize);
+                // send the slice of data off set by the index and of length len.
+                _spiDevice.Write(data.Slice(index, len));
+                // add the length just sent to the index
+                index += len;
+            } while (index < data.Length); // repeat until all data sent.
+        }
+
+        public void Dispose()
+        {
+            if(_gpioDevice != null)
```

Comment:
You don't need to check for null, that is what the question mark is doing before calling dispose. Also, don't you also need to dispose the spidevice? I know that one gets passed in, but the pattern we have been following in the bindings is that we dispose the Spi or I2c device when passed in. So basically the code of the dispose method should look like:

```cs
        public void Dispose()
        {
            _gpioDevice?.Dispose();
            _gpioDevice = null;
            _spiDevice?.Dispose();
            _spiDevice = null;
        }
```

Reply:
I thought the last change to the PR had implemented this for the GPIO device. I left out the SPI device from this pattern as I read the opinion that the caller should dispose as the caller created the device. ... Anyway happy to follow your lead. Will change this at the next opportunity.

Reply:
yeah I would prefer to have all of the bindings following the same pattern for now (disposing the devices) and if we then decide to change this, we'll go through all the bindings and change them in one PR, but ideally we want to aim for consistency 😉

Reply:
The latest PR should now include the changes to the dispose pattern. Hope it looks good. Should I be pressing the [Resolve conversation] button or is that for you to do ?

Reply:
@Frankenslag thanks! Personal preference - when I have few comments I usually let the comment author resolve them (and resolve them after a day if no one does it) - for larger number of comments I resolve them (there are usually a few comments which need more discussion - without resolving the noise it's hard to see them)